qiita_db/user.py

```diff
@@ -232,10 +233,15 @@ def create(cls, email, password, info=None):
         # create user default sample holder
         sql = ("INSERT INTO qiita.analysis "
                "(email, name, description, dflt, analysis_status_id) "
-               "VALUES (%s, %s, %s, %s, 1)")
+               "VALUES (%s, %s, %s, %s, 1) RETURNING analysis_id")
         conn_handler.add_to_queue(queue, sql,
                                   (email, '%s-dflt' % email, 'dflt', True))
 
+        portal_id = convert_to_id('QIITA', 'portal_type', 'portal')
```

Comment:
Ok, maybe 4 plus tests.

The important point is that you're modifying this in order to fix something, and you're letting the fix not completed and saying that this will go in another PR.

the way of working here is not that we are going to review half of your commits. We are going to review full functionality. You already had a lot of examples in how atomicity is accomplished in this type of complex issues. First, solving the main part, and then keep improving each of the files independently with complete changes. What you're doing here is not a good way of programming.

Either you remove all the changes you've done in this function and you do it in the next PR, or you implement all the changes in here.